internal/commands/build.go

```diff
@@ -302,3 +320,81 @@ func (c *OneBuildConfig) New() LoadableConfig {
 func OneBuildConfigFrom(cmd *cli.Command) (*OneBuildConfig, error) {
 	return ConfigFromCLI[*OneBuildConfig](oneBuildConfigKey, cmd)
 }
+
+func uploadToUrl(uploadUrl string, uploadBodyParams []shared.UploadBodyParams, filePath string) error {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return err
+	}
+	fileSize := fileInfo.Size()
+
+	var requestBody bytes.Buffer
+	multipartWriter := multipart.NewWriter(&requestBody)
+
+	for _, param := range uploadBodyParams {
+		_ = multipartWriter.WriteField(param.Key, param.Value)
+	}
+
+	fileWriter, err := multipartWriter.CreateFormFile("file", fileInfo.Name())
+	if err != nil {
+		return err
+	}
+
+	progressReader := &progressReader{
+		reader: file,
+		total:  fileSize,
+		callback: func(percentage float64, loaded int64, total int64) {
+			fmt.Printf("Upload progress: %.2f%% (%d/%d bytes)\r", percentage, loaded, total)
+		},
+	}
+
+	_, err = io.Copy(fileWriter, progressReader)
+	if err != nil {
+		return err
+	}
+	print("\n")
```

Comment:
Is this necessary with our logger?

Reply:
yea because I have `\r` in the preceding line, this prevents the next line from mixing characters with the upload progress line

Reply:
could do something where i'm like "if progress == 100% then use \n otherwise use \r" if you'd prefer

Reply:
It might be more concise having this line be printed by the processReader.

```go
progressReader := &progressReaderType{
	reader: file,
	total:  fileSize,
	callback: func(percentage float64, loaded int64, total int64, eof bool) {
		if !eof {
			logger.Info(fmt.Sprintf("Upload progress: %.2f%% (%d/%d bytes)\r", percentage, loaded, total))
		} else {
			logger.Info(fmt.Sprintf("Upload Complete: %d bytes\n", total))
		}
	},
}

func (pr *progressReaderType) Read(p []byte) (int, error) {
	n, err := pr.reader.Read(p)
	if err != nil && err == io.EOF {
		pr.callback(percentage, pr.read, pr.total, true)
	}
	if n > 0 {
		pr.read += int64(n)
		percentage := float64(pr.read) / float64(pr.total) * 100
		pr.callback(percentage, pr.read, pr.total, false)
	}
	return n, err
}
```

Reply:
ooh yea i like that